src/test/scala/uk/gov/nationalarchives/LambdaTest.scala

```diff
@@ -0,0 +1,225 @@
+package uk.gov.nationalarchives
+
+import cats.effect.IO
+import com.amazonaws.services.lambda.runtime.events.SQSEvent
+import com.amazonaws.services.lambda.runtime.events.SQSEvent.SQSMessage
+import com.github.tomakehurst.wiremock.WireMockServer
+import com.github.tomakehurst.wiremock.client.WireMock._
+import com.github.tomakehurst.wiremock.http.RequestMethod
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers._
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.s3.S3AsyncClient
+import software.amazon.awssdk.services.sfn.SfnAsyncClient
+import uk.gov.nationalarchives.FileProcessor.{TREInput, TREInputParameters}
+import uk.gov.nationalarchives.SeriesMapper.Output
+import upickle.default._
+
+import java.net.URI
+import java.util.UUID
+import scala.jdk.CollectionConverters._
+
+class LambdaTest extends AnyFlatSpec with BeforeAndAfterEach {
+  implicit val treInputWriter: Writer[TREInput] = macroW[TREInput]
+  implicit val treInputParamsWriter: Writer[TREInputParameters] = macroW[TREInputParameters]
+  implicit val sfnRequestReader: Reader[SFNRequest] = macroR[SFNRequest]
+  implicit val outputReader: Reader[Output] = macroR[Output]
+
+  case class SFNRequest(stateMachineArn: String, name: String, input: String)
+
+  val reference = "TEST-REFERENCE"
+  val uuids: List[String] = List(
+    "c7e6b27f-5778-4da8-9b83-1b64bbccbd03",
+    "61ac0166-ccdf-48c4-800f-29e5fba2efda",
+    "4e6bac50-d80a-4c68-bd92-772ac9701f14",
+    "c2e7866e-5e94-4b4e-a49f-043ad937c18a",
+    "27a9a6bb-a023-4cab-8592-39b44761a30a"
+  )
+
+  val metadataFiles: List[String] = List(
+    "asset-metadata.csv",
+    "bagit.txt",
+    "bag-info.txt",
+    "file-metadata.csv",
+    "folder-metadata.csv",
+    "manifest-sha256.txt",
+    "tagmanifest-sha256.txt"
+  )
+
+  override def beforeEach(): Unit = {
+    sfnServer.resetAll()
+    s3Server.resetAll()
+    sfnServer.start()
+    s3Server.start()
+  }
+
+  val s3Server = new WireMockServer(9003)
+  val sfnServer = new WireMockServer(9004)
+  val testOutputBucket = "outputBucket"
+
+  case class IngestParserTest() extends Lambda {
+    private val s3AsyncClient: S3AsyncClient = S3AsyncClient
+      .crtBuilder()
+      .endpointOverride(URI.create("http://localhost:9003"))
+      .region(Region.EU_WEST_2)
+      .build()
+
+    private val sfnAsyncClient: SfnAsyncClient = SfnAsyncClient
+      .builder()
+      .endpointOverride(URI.create("http://localhost:9004"))
+      .region(Region.EU_WEST_2)
+      .build()
+
+    override val s3: DAS3Client[IO] = DAS3Client[IO](s3AsyncClient)
+    override val sfn: DASFNClient[IO] = new DASFNClient(sfnAsyncClient)
+    override val seriesMapper: SeriesMapper = new SeriesMapper(Map("cite" -> "TEST SERIES"))
+    var count: Int = -1
+
+    override val uuidGenerator: () => UUID = () => {
```

Comment:
Could you apply the same iterator solution here as the one in the FileProcessorTest?